src/replicache.ts

```diff
@@ -926,6 +952,22 @@ export class Replicache<MD extends MutatorDefs = {}> {
     return {requestID, syncHead, ok: httpRequestInfo.httpStatusCode === 200};
   }
 
+  protected async _persist(): Promise<void> {
+    await this._ready;
+    const clientID = await this.clientID;
+    return this._persistLock.withLock(() =>
+      persist.persist(clientID, this._memdag, this._perdag),
```

Comment:
should this await?

Reply:
doesn't it?

Reply:
it handles locking of the dags internally, i was wondering if we should push the persistLock into persist.

Reply:
The persist lock should be per memdag/replicache instance. I don't see the benefit of doing this inside persist.ts given that.

Reply:
good point, agree.